### PR TITLE
Oh So Many Fixes!

### LINKED ISSUE #9: Auto-Install Grappelli
We should have django-grappelli auto setup in our new projects that are created. Likely this just means including it in a requirements.txt file and putting it in installed apps. May be a few other places to hook it in, not entirely sure.

https://github.com/sehmaschine/django-grappelli

### LINKED ISSUE #51: Vault Bug: Running `yurt vault` without anything in project folder
`yurt vault` will save a `vault_....json` file in the parent directory of the project directory if `Vagrantfile` isn't in project directory. Figure out a way to better manage vaultfile creation and overall file management

### LINKED ISSUE #52: Vault Bug: `yurt add_settings --vault` is broken
- We send a key-pair secret to the vault (which works)
- We put in a "lookup tag" that our lookup module should read (great)
- Ansible doesn't read the lookup tag because it's not quoted (no bueno)

### LINKED ISSUE #53: Setup Bug: `sudo ansible-galaxy` doesn't always work
We should also include directions OR have a fab command for setting up a `~/.ansible.cfg` file that defines a `roles_path` that is writable by the user (since default behavior is to write to the root-only  `/etc/ansible/roles`). Having Yurt do setup of a user's Ansible configuration stuff might automagically improve the flow for beginners, but for more experienced Ansible-using devs, automated Ansible setup could break their local setups.

### PR DESCRIPTION
**Pull request by paolopaolopaolo**
- Removes quotes from priv/pub key in base.yml
- Fixes #9: Autoinstall Grappelli
- Cleanup dependencies, enable ad-hoc calls for debugging
- Fixes #51: Adds `--dest` flag to vault, overrides needing Vagrantfile
- Fixes #52: Create method that automatically searches for Vagrantfile
- Fixes #53: Update setup instructions in README.rst
- Remove `$` from snippets in README.rst
- Add some notes on Vagrant usage to README.rst

@rmutter, Some fixes for polishing stuff. Skipping a few version numbers (because I deployed them to PyPi before finding out about TestPyPi). Let me know if it's good to merge and I can make a release.

### CODE REVIEW

File: yurt/fabfile/add.py [hunk 1]
```diff
@@ -108,18 +110,69 @@ def remote_server():
...
+    settings = {
+        'git_public_key': public_key,
+        'git_private_key': private_key,
+        'vagrant': {
+            'db_pw': generate_printable_string(15, False),
+            'secret_key': generate_printable_string(40)
+        }
+    }
+    if vault:
+        vagrantfile_dir = find_vagrantfile_dir()
```
Comment (rmutter) [inline yurt/fabfile/add.py]: to keep this method shorter I wonder if the code under `if vault:` should be put in a separate method

File: yurt/fabfile/add.py [hunk 2]
```diff
@@ -108,18 +110,69 @@ def remote_server():
...
+                                                        "lookup('vault', '{0}', 'private_key', '{1}')".format(unique_key,
+                                                                                                          path),
+                                                        "}}\\\""])
+                print("Add this public key to your SSH deploy keys in Github:\n{}".format(public_key))
+            else:
+                raise Exception('Vault is unavailable!')
+        except ConnectionError:
+            print('Cannot connect to vault: Connection Error')
+    if 'fabric_settings.py' in os.listdir('.'):
+        continue_process = raw_input('You already have `fabric_settings.py in this folder. Overwrite? (Y/N)')
```
Comment (rmutter) [inline yurt/fabfile/add.py]: missing a ` here?

File: yurt/fabfile/add.py [hunk 3]
```diff
@@ -108,18 +110,69 @@ def remote_server():
...
+            if client.is_authenticated() and not client.is_sealed():
+                unique_key = 'secret/git_keys_{}'.format(generate_printable_string(25, False))
+                client.write(unique_key, private_key=private_key, public_key=public_key)
+                settings['git_public_key'] = " ".join(["\\\"{{",
+                                                       "lookup('vault', '{0}', 'public_key', '{1}')".format(unique_key,
+                                                                                                            path),
+                                                       "}}\\\""])
+                settings['git_private_key'] = " ".join(["\\\"{{",
+                                                        "lookup('vault', '{0}', 'private_key', '{1}')".format(unique_key,
+                                                                                                          path),
```
Comment (rmutter) [inline yurt/fabfile/add.py]: `path` same indent as `unique_key`

File: yurt/requirements.txt [hunk 4]
```diff
@@ -1,3 +1,5 @@
 Django
 django_compressor
-psycopg2
\ No newline at end of file
+djangorestframework
+django-grappelli
```
Comment (rmutter) [inline yurt/requirements.txt]: is the setup for `django-grappelli` already in the app (i.e. appropriate code in settings.py, etc), we just forgot to put it in `requirements.txt`?

**Comment by rmutter** [comment]
@paolopaolopaolo couple of comments!

File: yurt/requirements.txt (hunk 4 above)
Comment (paolopaolopaolo) [inline yurt/requirements.txt]: @rmutter, Yup! I added the grappelli configs to `base.py` and `urls.py`.

**Comment by paolopaolopaolo** [comment]
@rmutter, Added a method in `utils.py` that handles writing values to the vault. 👍 ?

**Comment by rmutter** [comment]
@paolopaolopaolo 👍 ready to merge, different topic from this PR, but I'm wondering if yurt is at the size where it could benefit from some tests - not sure how hard those tests would be to write considering the nature of what the functionality is.

**Pull request merged**